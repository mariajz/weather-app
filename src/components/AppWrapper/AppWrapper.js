import React from 'react';
import { SafeAreaView, StatusBar, Platform } from 'react-native';

const AppWrapper = ({ children }) => {
    return (
        <>
            {Platform.OS === 'android' && (
                <StatusBar barStyle="light-content" backgroundColor="maroon" />
            )}
            <SafeAreaView>{children}</SafeAreaView>
        </>
    );
};

export default AppWrapper;
