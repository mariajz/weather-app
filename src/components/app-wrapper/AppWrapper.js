import React from 'react';
import { StatusBar, Platform } from 'react-native';
import { BackgroundImage } from '../visual-elements';

const AppWrapper = ({ children }) => {
    return (
        <>
            {Platform.OS === 'android' && (
                <StatusBar
                    barStyle="dark-content"
                    backgroundColor="transparent"
                />
            )}
            <BackgroundImage />
            {children}
        </>
    );
};

export default AppWrapper;
