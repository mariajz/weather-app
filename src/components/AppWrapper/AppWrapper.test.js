import { render } from '@testing-library/react-native';
import React from 'react';
import AppWrapper from './AppWrapper';

describe('AppWrapper', () => {
    it('should render SafeArea for android and ios', () => {
        const container = render(<AppWrapper />);

        expect(container).toMatchSnapshot();
    });
});
