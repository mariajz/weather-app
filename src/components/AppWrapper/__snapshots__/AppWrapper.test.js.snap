// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppWrapper should render SafeArea for android and ios 1`] = `<RCTSafeAreaView />`;
